src/backends/sentry_backend_breakpad.cpp

```diff
@@ -23,10 +22,19 @@ extern "C" {
 
 #ifdef SENTRY_PLATFORM_WINDOWS
 #    include "client/windows/handler/exception_handler.h"
+#elif defined SENTRY_PLATFORM_DARWIN
+#    include "client/mac/handler/exception_handler.h"
+#    include <sys/sysctl.h>
 #else
 #    include "client/linux/handler/exception_handler.h"
 #endif
 
+#ifdef SENTRY_PLATFORM_UNIX
+extern "C" {
+#include "sentry_sync.h"
```

Comment:
Is there a specific reason this was moved down below the breakpad includes? Would be nice to add a comment about why?

Reply:
In my code base I added support for macOS 10.6, which needs to emulate a recursive mutex, and the way the current apple code path manages setting up the mutex is all with defines, so the simplest solution was to just redefine the pthread routines to recursive variants, so it causes issues with some of the c++ includes that have inline code using pthread_mutext_init.

Ideally I would have refactored sentry_sync so it didn't have everything done via macros and instead implemented all the routines in the c files, but I was trying not to rewrite all of that unit.

Reply:
Oh, so you are saying that you have your own breakpad fork that overrides this here?:

https://github.com/getsentry/sentry-native/blob/cbd7b964b6721b622fb4eb44b77f719aa944e7d4/src/sentry_sync.h#L202

Can you solve that differently by specifying some custom CFLAGS? It sounds like quite a hack that is hard to get right, and also not a case I would like to optimize for really.

Reply:
effectively, I'm not sure how a CFLAG would help the situation. 
The implementation is basically the same principle that SDL has used for ages for recursive mutex support.

Reply:
the proper way would be to to clean up sentry_sync.h so its not all header based, as its rather fragile to begin with and still has lots of warnings depending on where its included.

Reply:
I think thats a lot of work, and also has some potential for breakage and perf regressions. But admittedly, it would be cleaner.